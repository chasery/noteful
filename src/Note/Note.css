.Note {
  display: flex;
  padding: 1rem;
  border-radius: 0.25rem;
  border: 0.0625rem solid #0000001c;
  list-style: none;
}

.Note + .Note {
  margin-top: 1rem;
}

.Note > div {
  flex-grow: 1;
}

.Note__title {
  margin: 0 0 1rem 0;
  font-size: 1.25rem;
}

.Note__modified {
  margin: 0;
  font-size: 1rem;
  font-weight: 400;
}

.Note__delete {
  flex-grow: 0;
  align-self: flex-end;
}
