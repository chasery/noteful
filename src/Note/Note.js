import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./Note.css";

class Note extends Component {
  render() {
    return (
      <li className="Note">
        <div>
          <h2 className="Note__title">
            <Link to={`/note/${this.props.id}`}>{this.props.name}</Link>
          </h2>
          <h3 className="Note__modified">
            Date modified on {this.props.modified}
          </h3>
        </div>
        <button className="Note__delete">Delete Note</button>
      </li>
    );
  }
}

export default Note;
