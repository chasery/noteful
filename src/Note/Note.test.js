import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Note from "./Note";

describe("Note component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<Note />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
