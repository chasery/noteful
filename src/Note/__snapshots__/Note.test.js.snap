// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders without error 1`] = `
<li
  className="Note"
>
  <div>
    <h2
      className="Note__title"
    >
      <Link
        to="/note/undefined"
      />
    </h2>
    <h3
      className="Note__modified"
    >
      Date modified on 
    </h3>
  </div>
  <button
    className="Note__delete"
  >
    Delete Note
  </button>
</li>
`;
